### PR TITLE
Fix type definitions for connect

### PR DESCRIPTION
**Pull request by anion155**

### CODE REVIEW

**Comment by anion155** [comment]
Will delete my rep in couple days

File: src/preact-redux.d.ts [hunk 1]
```diff
@@ -20,16 +20,16 @@ export interface DispatchProp<S> {
 }
 
 interface AdvancedComponentDecorator<TProps, TOwnProps> {
-    (component: AnyComponent<TProps, {}>): ComponentConstructor<TOwnProps, {}>;
+    (component: ComponentFactory<TProps>): ComponentConstructor<TOwnProps, {}>;
```
Comment (developit) [inline src/preact-redux.d.ts]: Would you be able to shed some light on what issue this solves?

**Review by developit — COMMENTED** [review]
Sorry for the slow review. I still haven't used TypeScript much so I'm not a great reviewer for definition changes.

File: src/preact-redux.d.ts (hunk 1 above)
Comment (anion155) [reply src/preact-redux.d.ts]: AnyComponent is for Component instances and ComponentFactory for factories. At least it was like so, previously.

**Pull request closed without merging**